milestone2.ts

```diff
@@ -83,32 +83,42 @@ const userData: Volunteer[] = [
   Hint: user data is stored in the userData object above. 
 */
 type GetNumber = (data: Volunteer[]) => number;
-let findAverage: GetNumber; // Define lambda function here
+let findAverage: GetNumber = (data) => {
+  return data.reduce((prev, curr) => prev + curr.age, 0) / data.length;
+};
 
-//console.log(findAverage(userData));
+// console.log(findAverage(userData));
 
 // Question 2: Data Handling
 /* Use data handling function(s) to find the first index of someone from San Francisco (SF).
   Return -1 if no one is from San Francisco.
   Hint: Use a lambda function as a value
 */
-let findIndexAns: GetNumber; // Code here
+let findIndexAns: GetNumber = (data) => {
+  return data.findIndex(x => x.city === "SF");
+};
 
-//console.log(findIndexAns(userData));
+// console.log(findIndexAns(userData));
 
 // Question 3: Filtering data
 /* Use data handling function(s) to find all of the volunteers from California (CA) over an age threshold n
  */
 type GetVolunteers = (data: Volunteer[], minAge: number) => Volunteer[];
-let findCAOverN: GetVolunteers; // Code here
+let findCAOverN: GetVolunteers = (data, minAge) => {
+  return data
+        .filter(x => x.state === "CA")
+        .filter(x => x.age >= minAge);
```

Comment:
This can be combined into one line with a logical and (&&)
```suggestion
        .filter(x => x.state === "CA" && x.age >= minAge)
```